docs/src/modules/spring/pages/actions-as-controller.adoc

```diff
@@ -1 +1,112 @@
 = Actions as Controllers
+
+Actions can be used to implement link:https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller[MVC Controllers] by
+acting as the external interface of a service, receiving requests, operating over the requests values and forwarding the call
+to other components in the same service.
+
+To illustrate how you can use an Action as a Controller, we will build on top of the xref:structuring-your-application.adoc[Value Entity Shopping Cart] example, adding a new Action to the existing shopping cart service.
+
+== Forwarding Commands
+
+The `forward` effect allows us to transform or further validate an incoming request before passing it on to another
+component and have the response message directly passed back to the client making the request. The response from the
+forwarded operation must have the same response type as the original command.
+
+In this example we accept requests with the same structure as the entity, by receiving a `LineItem`, but add some additional verification of the
+request and only conditionally forward the request to the entity if the verification is successful:
+
+
+.src/main/java/com/example/api/ShoppingCartController.java
+[source,java]
+----
+include::example$spring-valueentity-shopping-cart/src/main/java/com/example/api/ShoppingCartController.java[tag=forward]
+----
+<1> `KalixClient` is injected on the constructor. It will be used to build calls to the underlining Entity.
+<2> Expose the command handler as a `POST` endpoint at specified path.
+<3> Check if the added item is carrots.
+<4> If it is "carrots" immediately return an error, disallowing adding the item.
+<5> For allowed requests, use `kalixClient` to get a `DeferredCall`.
+<6> The `DeferredCall` is used with `effects().forward()` to forward the request to the entity.
+
+NOTE: You might be wondering what the `kalixClient` is about. For now, think of it as a lightweight HTTP client allowing you to reach out to other Kalix services. All details can be found at xref:call-another-service.adoc[] chapter.
+
+== Transform Request and Response to Another Component
+
+The `asyncReply` and `asyncEffect` effects allow us to process and transform a request before calling another component and then also transform the response.
+
+As an example, let us look at the problem of creating a new entity with an id generated by the service itself.
```

Comment:
```suggestion
As an example, let us look at the problem of creating a new entity with an id generated by another component.
```

Reply:
for the record, I did this change but also for the java counterpart